crates/uv/src/commands/python/install.rs

```diff
@@ -291,42 +292,102 @@ pub(crate) async fn install(
                     .or_default()
                     .push(target.clone());
             }
-            Err(uv_python::managed::Error::LinkExecutable { from, to, err })
+            Err(uv_python::managed::Error::LinkExecutable { from: _, to, err })
                 if err.kind() == ErrorKind::AlreadyExists =>
             {
-                // TODO(zanieb): Add `--force`
-                if reinstall {
-                    fs_err::remove_file(&to)?;
-                    installation.create_bin_link(&target)?;
-                    debug!(
-                        "Updated executable at {} to {}",
-                        target.user_display(),
-                        installation.key(),
-                    );
-                    changelog.installed.insert(installation.key().clone());
-                    changelog
-                        .installed_executables
-                        .entry(installation.key().clone())
-                        .or_default()
-                        .push(target.clone());
-                    changelog.uninstalled_executables.insert(target);
-                } else {
-                    if !is_same_file(&to, &from).unwrap_or_default() {
-                        errors.push((
+                debug!(
+                    "Inspecting existing executable at {}",
+                    target.user_display()
+                );
+
+                // Figure out what installation it references, if any
+                let existing = find_matching_bin_link(&existing_installations, &target);
+
+                match existing {
+                    None => {
+                        // There's an existing executable we don't manage, require `--force`
+                        if !force {
+                            errors.push((
+                                installation.key(),
+                                anyhow::anyhow!(
+                                    "Executable already exists at `{}` but is not managed by uv; use `--force` to replace it.",
+                                    to.user_display()
+                                ),
+                            ));
+                            continue;
+                        }
+                        debug!(
+                            "Replacing existing executable at {} due to `--force`",
+                            target.user_display()
```

Comment:
We may want to use `simplified_display` for these which is always absolute, but don't feel strongly.